pkg/node/kubelet.go

```diff
@@ -47,17 +86,50 @@ func StartKubelet(cfg *config.MicroshiftConfig) error {
 		args = append(args, "--log-file="+filepath.Join(cfg.LogDir, "kubelet.log"))
 	}
 
-	if err := command.ParseFlags(args); err != nil {
-		logrus.Fatalf("failed to parse flags:%v", err)
+	cmd := &cobra.Command{
+		Use:          "kubelet",
+		Long:         `kubelet`,
+		SilenceUsage: true,
+		RunE:         func(cmd *cobra.Command, args []string) error { return nil },
 	}
-	logrus.Infof("starting kubelet %s, args: %v", cfg.HostIP, args)
 
-	go func() {
-		command.Run(command, args)
-		logrus.Fatalf("kubelet exited")
-	}()
+	cleanFlagSet := pflag.NewFlagSet(componentKubelet, pflag.ContinueOnError)
+	cleanFlagSet.SetNormalizeFunc(cliflag.WordSepNormalizeFunc)
+	kubeletFlags := kubeletoptions.NewKubeletFlags()
+	kubeletFlags.AddFlags(cleanFlagSet)
+	kubeletoptions.AddKubeletConfigFlags(cleanFlagSet, kubeletConfig)
+	kubeletoptions.AddGlobalFlags(cleanFlagSet)
 
+	if err := cmd.ParseFlags(args); err != nil {
+		logrus.Fatalf("failed to parse flags:%v", err)
+		return err
+	}
+	s.kubeletflags = kubeletFlags
+	s.kubeconfig = kubeletConfig
+	s.kubeconfigfile = filepath.Join(cfg.DataDir, "resources", "kubelet", "kubeconfig")
+	logrus.Infof("Starting kubelet %s, args: %v", cfg.HostIP, args)
 	return nil
+
+}
+
+func (s *Kubelet) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+	defer close(stopped)
+
+	// construct a KubeletServer from kubeletFlags and kubeletConfig
+	kubeletServer := &kubeletoptions.KubeletServer{
+		KubeletFlags:         *s.kubeletflags,
+		KubeletConfiguration: *s.kubeconfig,
+	}
+
+	kubeletDeps, err := kubelet.UnsecuredDependencies(kubeletServer, utilfeature.DefaultFeatureGate)
+	if err != nil {
+		logrus.Fatalf("Error in fetching depenedencies %v", err)
+	}
+	if err := kubelet.Run(ctx, kubeletServer, kubeletDeps, utilfeature.DefaultFeatureGate); err != nil {
```

Comment:
add a readiness probe (similar to kube-controller-manger or kube-scheduler), using kubelet healthz option `healthzPort` https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/